src/toggle-button/index.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+
+import { getBaseProps } from '../internal/base-component';
+import useBaseComponent from '../internal/hooks/use-base-component';
+import { applyDisplayName } from '../internal/utils/apply-display-name';
+import { ToggleButtonProps } from './interfaces';
+import { InternalToggleButton } from './internal';
+
+export { ToggleButtonProps };
+
+const ToggleButton = React.forwardRef(
+  (
+    {
+      children,
+      iconName,
+      pressedIconName,
+      iconUrl,
+      pressedIconUrl,
+      iconSvg,
+      pressedIconSvg,
+      iconAlt,
+      variant = 'normal',
+      loading = false,
+      loadingText,
+      disabled = false,
+      disabledReason,
+      wrapText = true,
+      ariaLabel,
+      ariaDescribedby,
+      onClick,
+      onFollow,
+      ariaControls,
+      fullWidth,
+      pressed = false,
+      onChange,
+      ...props
+    }: ToggleButtonProps,
+    ref: React.Ref<ToggleButtonProps.Ref>
+  ) => {
+    const baseComponentProps = useBaseComponent('ToggleButton', {
+      props: { fullWidth, iconName, pressedIconName, pressed, variant, wrapText },
+    });
+    const baseProps = getBaseProps(props);
+
+    return (
+      <InternalToggleButton
+        {...baseProps}
+        {...baseComponentProps}
+        ref={ref}
+        iconName={iconName}
```

Comment:
Have we considered setting the `fill` property of the SVG on our end instead of requiring the `pressed` icon counterpart? This would not only make its usage easier but also reduce the risk of erroneous usage from customers.

And we could constrain the set of icon names that can be used to a few that we know would work well with this approach (this feature only makes sense with a very reduced number of icons anyway).

Reply:
Yes, we considered giving customers more control over this component in the doc `quip/ZwfaAocOlIHK/API-Proposal-Toggle-button#temp:C:LXedc371330557f46ffb874e0dc5` but decided to introduce new states for pressed icons instead to avoid possible confusions. 

> And we could constrain the set of icon names that can be used to a few that we know would work well with this approach (this feature only makes sense with a very reduced number of icons anyway).

yes, it's already described in the doc `quip/wWzGA4MOBt9D/Component-article-Toggle-Button#temp:C:KZU2646763fff38442aad70540b9`

Reply:
My concern is that some customers might use the component e.g, like this:

```
<ToggleButton pressed={pressed} iconName="star" onChange={...}>Favourite</ToggleButton>
```

without realizing they also need to set `iconNamePressed` (or `iconSVGPressed` or `iconURLPressed`). Until they manually interact with it, the component will good to them and TypeScript will not yield any error either.

So the question is whether the component could actually be used like that without the `pressed` props, e.g, by us internally adding the `-filled` icon to the icon name, or by setting the `fill` attribute in the SVG.

Reply:
This behavior will be described in the usage guideline for this component (currently drafted in the doc `quip/wWzGA4MOBt9D/Component-article-Toggle-Button#temp:C:KZU2646763fff38442aad70540b9`)

> So the question is whether the component could actually be used like that without the pressed props, e.g, by us internally adding the -filled icon to the icon name, or by setting the fill attribute in the SVG.

We considered this option, but came to realize that we have no consistency over names in our iconography, e.g. filled icon might have either `-filled` or `-action` postfix, so we cannot rely on them.

Reply:
> We considered this option, but came to realize that we have no consistency over names in our iconography, e.g. filled icon might have either -filled or -action postfix, so we cannot rely on them.

But we have control over which icon names to use internally, and we could constrain the accepted icon names to a subset of those which have a filled version.

Anyway, if that decision is already taken, what about at least issuing dev warnings if the corresponding `pressed` icon is not defined?

Reply:
Sounds reasonable. Added

Reply:
Thanks. Now you just need to add coverage to those warnings to let the coverage check pass :)